worker/handler/handler.go

```diff
@@ -277,6 +338,21 @@ func (h *Handler) RunStart() (ch *sb.Channel, err error) {
 
 	// create sandbox if needed
 	if h.sandbox == nil {
+		// if estimated mem usage exceeds the hard limit, we would block 
+		// all the creation requests until there are more free mem.
+		hms.sbCreatingLock.Lock()
+		for hms.sbHardLimit != 0 && (getMemUsage() + (hms.numSbCreating + 1) * hms.avgHandlerMemUsage > hms.sbHardLimit) {
+			hms.sbCreatingLock.Unlock()
+			select {
+			case <-hms.sbCreatingChan:
+			case <-time.After(4 * time.Second):
```

Comment:
4 seconds is an eternity...

Reply:
This is replaced by polling in the separate scheduling goroutine in the new commit.